engine/classes/Elgg/Database/MetastringsTable.php

```diff
@@ -58,6 +60,43 @@ function getId($string, $case_sensitive = true) {
 			return $this->getIdCaseInsensitive($string);
 		}
 	}
+
+	/**
+	 * Get a map of metastring identifiers to string values (case sensitively)
+	 *
+	 * @param string[] $string_keys Strings to look up
+	 *
+	 * @return int[] map of [string] => [id]
+	 */
+	function getMap(array $string_keys) {
+		if (!$string_keys) {
+			return [];
+		}
+		if (count($string_keys) === 1) {
+			$key = reset($string_keys);
+			return [$key => $this->getIdCaseSensitive($key)];
+		}
+
+		$missing = array_fill_keys($string_keys, true);
+
+		$set_element = array_map(function ($string) {
+			return "BINARY '" . $this->db->sanitizeString($string) . "'";
+		}, $string_keys);
+		$set = '(' . implode(',', $set_element) . ')';
```

Comment:
This would be easier to read if you moved the parenthesis to the query string below.